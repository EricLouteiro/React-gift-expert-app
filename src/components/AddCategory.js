import React, { useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategory}) => {

    const [inputValue, setInputValue] = useState('')
    
    const handleInputValue = ( e ) => {
        setInputValue( e.target.value );

        console.log('handle input change llamado')
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategory( cats => [ inputValue, ...cats ]);
            setInputValue('');            
        }
    }
    

    return (
        <form onSubmit={ handleSubmit }>
            <p> { inputValue } </p>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputValue }
            /> 

        </form>
        
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
