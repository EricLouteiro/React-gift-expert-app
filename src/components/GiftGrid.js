// import React, { useState, useEffect } from 'react'
import { useFetchGifts } from '../hooks/useFetchGifts';
// import { getGift } from '../helpers/getGifts';
import { GiftGridItem } from './GiftGridItem';
import PropTypes from 'prop-types'

export const GiftGrid = ({category}) => {

    // const [images, setImages] = useState([]);
    
    const { data: images, loading } = useFetchGifts( category );
    
    // useEffect(() => {
    //     getGift( category ).then( setImages );
    // }, [ category ]);
    
    
    // getGift();
    
    return (
        <>
            <h3>{category}</h3>

            { loading && <p className='animate__animated animate__zoomIn'>Loading...</p>}
            <div className='card-grid'>
                {
                    images.map( img => (
                        <GiftGridItem 
                            key={img.id}
                            { ...img } 
                        />
                    ))
                }
            </div>
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}