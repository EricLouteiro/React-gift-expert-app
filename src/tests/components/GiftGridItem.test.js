import React from 'react'
import { shallow } from 'enzyme';
import { GiftGridItem } from '../../components/GiftGridItem';

describe('Pruebas en giftGridItem', () => {
    

    const title = 'Un título';
    const url = 'http://localhost.com';
    let wrapper = shallow( 
        <GiftGridItem 
            url={url}
            title={title}    
        />);

    test('debe de mostrar el componente correctamente', () => {
       

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de tener un párrafo con el titulo', () => {
        
        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );

    });

    test('debe de tener la imagen igual al url y alt ', () => {
        
        const img = wrapper.find('img');

        console.log(img)
        

        expect( img.prop('src') ).toBe( url );
        expect( img.prop( 'alt') ).toBe( title );
    });

    test('debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');

        // console.log();

        expect( div.hasClass('animate__fadeInDown') ).toBe( true );
    });
    
    
});
