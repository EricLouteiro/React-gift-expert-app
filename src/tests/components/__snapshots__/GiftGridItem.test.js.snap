// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en giftGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Un título"
    src="http://localhost.com"
  />
  <p>
    Un título
  </p>
</div>
`;
